contracts/AppLogic.sol

```diff
@@ -53,9 +50,14 @@ contract AppLogic is SuperAppBase, Initializable {
             )
         );
         
-        acceptedToken = ISuperToken(_acceptedToken);
         locker = ILocker(_locker);
-        minFlowRate = _minFlowRate;
+        acceptedToken = ISuperToken(locker.tokenAddress);
+        syncPrice();
+    }
+
+    // calculating minFlowRate based on lock price divided by duration
+    function syncPrice() public {
+        minFlowRate = locker.keyPrice / locker.expirationDuration; 
```

Comment:
🚨 expirationDuration can be zero.

Reply:
https://github.com/unlock-protocol/unlock/blob/3ee617d947118cf9d68cacab19649664f84ce722/packages/contracts/src/contracts/PublicLock/PublicLockV11.sol#L1128